src/components/toast/toast.component.js

```diff
@@ -128,6 +145,7 @@ const Toast = React.forwardRef(
             {...tagComponent(restProps["data-component"] || "toast", restProps)}
             {...toastProps}
             ref={toastContentNodeRef}
+            tabIndex={hasBeenFocused ? -1 : 0}
```

Comment:
strictly speaking this ternary isn't needed - tabIndex -1 still allows focus to be given programatically, while not including it in the natural keyboard tab order. However I recall there were issues with cypress tests working with tabIndex -1 - if you haven't already, it might be worth testing with just `tabIndex="-1"` here and leaving it with that if there are no issues, as that's much simpler/better code.

A further benefit would be that it we don't use `hasBeenFocused` to control a rendered prop then it can become a ref rather than state, and cause fewer rerenders. I suspect Cypress will say "no" but worth a try if you haven't already done so (which you may well have, in which case ignore this!)

Reply:
I on purposely avoided doing any tabIndex tests in Cypress as there is a known bug with the `cypress-plugin-tab` package with trying to test `tabIndex={-1}`. Hence, I used the CSS value for testing focus. Not ideal but best I could think of without testing the tabIndex directly.